### PR TITLE
Some deployment notes

### PR DESCRIPTION
**Pull request by wlach**
### Pull Request checklist

- [x] The pull request has a descriptive title (and a reference to an issue it
      fixes, if applicable)
- [x] All tests and linter checks are passing
- [x] The pull request is free of merge conflicts

### CODE REVIEW

**Comment by wlach** [comment]
@Iinh Can you take a look at this? I'd like you to deploy the next release just to be sure someone who's not me can also do this if need be.

File: README.md [hunk 1]
```diff
@@ -69,6 +69,23 @@ following command:
...
+(https://dictionary.telemetry.mozilla.org) is deployed from the `production`
+branch on this repository, which usually corresponds to the latest GitHub
+release. To update the Glean Dictionary to the latest version, follow this
+procedure:
+
+- Create a new release (you might find
+  [this iodide notebook](https://alpha.iodide.io/notebooks/6616/) useful for
+  creating a set of notes on what changed)
+- From a local checkout, update the `production` branch to be in sync with the
+  tag you just created, then push to the production branch.
```
Comment (Iinh) [inline README.md:84]: is the production branch automatically deployed to dictionary.telemetry.mozilla.org or do I need to ask anyone on SRE to deploy it?

File: README.md [hunk 2]
```diff
@@ -69,6 +69,23 @@ following command:
 npm run test:jest -- -u
 ```
 
+## Deployment
+
+The production version of the Glean Dictionary
+(https://dictionary.telemetry.mozilla.org) is deployed from the `production`
+branch on this repository, which usually corresponds to the latest GitHub
+release. To update the Glean Dictionary to the latest version, follow this
```
Comment (Iinh) [inline README.md:77]: Is `main` the target branch when we [create a Github release](https://github.com/mozilla/glean-dictionary/releases/new)?

**Review by Iinh — APPROVED** [review]
Thanks @wlach!

File: README.md (hunk 2 above)
Comment (wlach) [reply README.md:77]: Yup, I added some clarification on this.

File: README.md (hunk 1 above)
Comment (wlach) [reply README.md]: It will be deployed automatically, added some clarification.

File: README.md (hunk 2 above)
Comment (Iinh) [reply README.md:77]: Thanks!

**Pull request merged**